decawave_ble/__init__.py

```diff
@@ -1,6 +1,20 @@
 import bluepy.btle
 import bitstruct
+import tenacity
 import json
+import logging
+
+logger = logging.getLogger(__name__)
+
+retry_exponential_wait_multiplier = 0.1
+retry_num_attempts = 3
+exponential_retry = tenacity.retry(
+        stop = tenacity.stop_after_attempt(retry_num_attempts),
+        wait = tenacity.wait_exponential(multiplier=retry_exponential_wait_multiplier),
+        before = tenacity.before_log(logger, logging.DEBUG),
```

Comment:
Should make the log level configurable in case debug is too chatty.

Reply:
Hmmm...I think I don't understand how the logging works, then. I thought that the parent logger established what to _do_ with the logs of various levels (should we handle them at all, where should we route them, how should we format them, etc.) whereas the log levels within the `tenacity` call are specifying what log level to _assign_ to each type of message from `tenacity`. I assigned `DEBUG` to the `before` and `after` messages precisely to make it easy to ignore them at the parent level.

Reply:
For now, the root loggers in the test functions are, in fact, set to `DEBUG`, but that's just for testing. I assumed that's where we'd change the log level to make it less chatty.